### PR TITLE
[Bindless][Exp] Add interop memory mapping to USM.

### PR DESCRIPTION
**Pull request by przemektmalon**
This patch introduces `urBindlessImagesMapExternalLinearMemoryExp` to allow mapping interop memory to USM regions.

### CODE REVIEW

**Comment by przemektmalon** [comment]
Associated [DPC++ MR](https://github.com/intel/llvm/pull/14701)

**Review by isaacault — APPROVED** [review]

**Review by DBDuncan — APPROVED** [review]

**Comment by przemektmalon** [comment]
Friendly ping @oneapi-src/unified-runtime-native-cpu-write @oneapi-src/unified-runtime-opencl-write @oneapi-src/unified-runtime-cuda-write @oneapi-src/unified-runtime-level-zero-write @oneapi-src/unified-runtime-maintain @MartinWehking

File: source/adapters/level_zero/image.cpp [hunk 1]
```diff
@@ -1074,6 +1074,20 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesMapExternalArrayExp(
...
+UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesMapExternalLinearMemoryExp(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice, uint64_t offset,
+    uint64_t size, ur_exp_interop_mem_handle_t hInteropMem, void **phRetMem) {
+  std::ignore = hContext;
+  std::ignore = hDevice;
+  std::ignore = size;
+  std::ignore = offset;
+  std::ignore = hInteropMem;
+  std::ignore = phRetMem;
+  logger::error(logger::LegacyMessage("[UR][L0] {} function not implemented!"),
```
Comment (pbalcer) [inline source/adapters/level_zero/image.cpp:1086]: No need for `logger::LegacyMessage` in any new logger uses. It exists only to maintain compatibility with sycl tests that expected an exact log output.

**Review by omarahmed1111 — APPROVED** [review]
OpenCL changes LGTM

**Comment by omarahmed1111** [comment]
@oneapi-src/unified-runtime-native-cpu-write @oneapi-src/unified-runtime-cuda-write Could you please review this when available ASAP, so we could include it in the next UR bump? Thanks!

File: source/adapters/level_zero/image.cpp (hunk 1 above)
Comment (DBDuncan) [reply source/adapters/level_zero/image.cpp]: Removed.

**Review by cppchedy — APPROVED** [review]

**Review by uwedolinsky — APPROVED** [review]
NativeCPU changes 👍

**Review by MartinWehking — APPROVED** [review]
LGTM

**Pull request merged**